File: results/preliminary_analysis/baseline_cpp_results/instruct_generations/cpp/files_before_feedback/question-103-pass-1.cpp
#include <bits/stdc++.h> 
 using namespace std ;
 void PythagoreanTriplet ( int n ) {
// Initialize variables
    int a = 0, b = 0, c = 0;

    // Loop through all possible values of a
    for (int a = 1; a <= n; a++) {
        // Loop through all possible values of b
        for (int b = 1; b <= n; b++) {
            // Calculate the value of c
            int c = sqrt(a*a + b*b);

            // Check if c is an integer and if the sum of a, b, and c is equal to n
            if (c == (int)c && a + b + c == n) {
                // If the conditions are met, print the triplet
                cout << a << " " << b << " " << c << endl;
            }
        }
    }
}

File: results/fine_tuning/test_orig_files/question-103-pass-1.py
from math import sqrt



def check(X, K):
    if X < K or K < 1:
        return False
    if X == K:
        return True
    for i in range(2, int(sqrt(X)) + 1):
        if X % i == 0 and check(i, K - 1):
            return True
    return False
